Source/WebCore/html/ImageBitmap.cpp

```diff
@@ -866,8 +866,12 @@ void ImageBitmap::createPromise(ScriptExecutionContext& scriptExecutionContext,
     auto alphaPremultiplication = alphaPremultiplicationForPremultiplyAlpha(options.premultiplyAlpha);
     if (sourceRectangle.returnValue().location().isZero() && sourceRectangle.returnValue().size() == imageData->size() && sourceRectangle.returnValue().size() == outputSize && options.orientation == ImageBitmapOptions::Orientation::None) {
         bitmapData->putPixelBuffer(imageData->pixelBuffer(), sourceRectangle.releaseReturnValue(), { }, alphaPremultiplication);
-        
-        auto imageBitmap = create(ImageBitmapBacking(WTFMove(bitmapData)));
+
+        OptionSet<SerializationState> serializationState = { SerializationState::OriginClean };
+        if (alphaPremultiplication == AlphaPremultiplication::Premultiplied)
+            serializationState.add(SerializationState::PremultiplyAlpha);
+
+        auto imageBitmap = create(ImageBitmapBacking(WTFMove(bitmapData), serializationState));
```

Comment:
Does it need the same thing of the similar case below, when the src and dst are different sizes? Maybe also test that :)

Reply:
I was hoping you or Said would tell me! Looking at it some more, I think we hit that second code path when the source and destination image buffers are different (orientation, size, etc.) Ashley's test case was for the like-to-like case, but we should handle the other, too. I think I see a way to test that.